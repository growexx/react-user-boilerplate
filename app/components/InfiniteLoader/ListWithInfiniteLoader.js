/* eslint-disable no-plusplus */
/**
 *
 * Demo with List having Infinite Loader.
 *
 */

import React from 'react';
import { List, message, Avatar, Skeleton } from 'antd';
import InfiniteScroll from 'react-infinite-scroller';
import request from 'utils/request';
import { ListWithInfiniteLoader as StyledList } from './StyledList';
import { API_ENDPOINTS } from '../../containers/Auth/constants';

class ListWithInfiniteLoader extends React.Component {
  state = {
    data: [],
    list: [],
    loading: false,
    hasMore: true,
  };

  componentDidMount() {
    this.fetchData(res => {
      this.setState({
        data: res.results,
        list: res.results,
        loading: false,
      });
    });
  }

  fetchData = callback => {
    request(API_ENDPOINTS.LIST, {
      method: 'GET',
    }).then(res => callback(res));
  };

  handleInfiniteOnLoad = () => {
    const { data } = this.state;
    this.setState({
      loading: true,
      list: data.map(singleData => ({ loading: true, ...singleData })),
    });
    if (data.length > 14) {
      message.warning('Infinite List loaded all');
      this.setState({
        hasMore: false,
        loading: false,
      });
      return;
    }
    this.fetchData(res => {
      const listData = data.concat(res.results);
      this.setState({
        data: listData,
        list: listData,
        loading: false,
      });
    });
  };

  render() {
    return (
      <StyledList>
        <div className="demo-infinite-container">
          <InfiniteScroll
            initialLoad={false}
            pageStart={0}
            loadMore={this.handleInfiniteOnLoad}
            hasMore={!this.state.loading && this.state.hasMore}
            useWindow={false}
          >
            <List
              dataSource={this.state.list}
              renderItem={item => (
                <List.Item key={item.id}>
                  <Skeleton avatar title={false} loading={item.loading} active>
                    <List.Item.Meta
                      avatar={<Avatar src={API_ENDPOINTS.LIST_AVATAR} />}
                      title={item.name.last}
                      description={item.email}
                    />
                  </Skeleton>
                </List.Item>
              )}
            />
          </InfiniteScroll>
        </div>
      </StyledList>
    );
  }
}

export default ListWithInfiniteLoader;
