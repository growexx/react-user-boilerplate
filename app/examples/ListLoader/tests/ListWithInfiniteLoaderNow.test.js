import React from 'react';
import { render, wait } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import request from 'utils/request';
import configureStore from '../../../configureStore';
import ListWithInfiniteLoader from '../ListWithInfiniteLoader';
let store;
jest.mock('utils/request');
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <ListWithInfiniteLoader />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );
describe('<ListWithInfiniteLoader />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    request.mockImplementation(() => Promise.resolve({ status: 1 }));
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('should render and match the snapshot', async () => {
    request.mockImplementation(() =>
      Promise.resolve({
        status: 1,
        results: [
          {
            name: {
              last: 'testInfiniteLoader',
              email: 'test@234.com',
            },
          },
        ],
      }),
    );
    const {
      container: { firstChild },
    } = componentWrapper();
    await wait(() => expect(request).toHaveBeenCalledTimes(2));
    expect(firstChild).toMatchSnapshot();
  });
});
