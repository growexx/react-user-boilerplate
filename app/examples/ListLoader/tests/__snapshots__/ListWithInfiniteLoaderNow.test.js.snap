// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListWithInfiniteLoader /> should render and match the snapshot 1`] = `
.c0 .demo-infinite-container {
  overflow: auto;
  border-radius: 4px;
}

.c0 .demo-loading-container {
  position: absolute;
  bottom: 40px;
  width: 100%;
  text-align: center;
}

<div
  class="c0"
>
  <div
    class="demo-infinite-container"
  >
    <div>
      <div
        class="ant-list ant-list-split"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-list-empty-text"
            >
              <div
                class="ant-empty ant-empty-normal"
              >
                <div
                  class="ant-empty-image"
                >
                  <svg
                    class="ant-empty-img-simple"
                    height="41"
                    viewBox="0 0 64 41"
                    width="64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      transform="translate(0 1)"
                    >
                      <ellipse
                        class="ant-empty-img-simple-ellipse"
                        cx="32"
                        cy="33"
                        rx="32"
                        ry="7"
                      />
                      <g
                        class="ant-empty-img-simple-g"
                        fill-rule="nonzero"
                      >
                        <path
                          d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                        />
                        <path
                          class="ant-empty-img-simple-path"
                          d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
                <div
                  class="ant-empty-description"
                >
                  No Data
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ListWithInfiniteLoader /> should render and match the snapshot 2`] = `
.c0 .demo-infinite-container {
  overflow: auto;
  border-radius: 4px;
}

.c0 .demo-loading-container {
  position: absolute;
  bottom: 40px;
  width: 100%;
  text-align: center;
}

<div
  class="c0"
>
  <div
    class="demo-infinite-container"
  >
    <div>
      <div
        class="ant-list ant-list-split"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <ul
              class="ant-list-items"
            >
              <li
                class="ant-list-item"
              >
                <div
                  class="ant-list-item-meta"
                >
                  <div
                    class="ant-list-item-meta-avatar"
                  >
                    <span
                      class="ant-avatar ant-avatar-circle ant-avatar-image"
                    >
                      <img
                        src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                      />
                    </span>
                  </div>
                  <div
                    class="ant-list-item-meta-content"
                  >
                    <h4
                      class="ant-list-item-meta-title"
                    >
                      testInfiniteLoader
                    </h4>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
